server/src/domain/types.rs

```diff
@@ -167,6 +167,82 @@ impl ValueType for UserId {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(from = "String")]
+pub struct DisplayName(String);
+
+impl DisplayName {
+    pub fn new(display_name: &str) -> Self {
+        Self(display_name.to_string())
+    }
+
+    pub fn as_str(&self) -> &str {
```

Comment:
I think instead of those two, we want conversion functions from `DisplayName` to `Option<&str>` and `Option<String>`. We want to translate from `""` to `None`

Reply:
Similarly, you might need `From<Option<String>>`